src/meta/node/src/server.rs

```diff
@@ -822,8 +843,9 @@ pub async fn start_service_as_election_leader(
                     },
                 }
             },
-        )
-        .await;
+        );
+    started::set();
```

Comment:
Can we check whether the server has started by connecting to the port like risedev does? Or we add some retry policy when other nodes try to connect to meta?

In some corner cases the current way still cannot ensure meta service starts before others. For example,
```
meta: started::set()
CN: start and connect
meta: server.await
```

Reply:
I guess the title of the PR might be misleading.

We don't have to verify that the service has actually started; confirming a successful binding to the listen address should suffice. This is because we're to resolve #11169, while the startup order of meta node and compute node does not matter since there's a retry mechanism.

Reply:
But I think we are only able to bind the address on `server.await`. After `TcpIncoming::new` is returned, the address is not bound yet.

```
#[tokio::main]
async fn main() {
    let socket1 = tonic::transport::server::TcpIncoming::new("127.0.0.1:8080".parse().unwrap(), true, None);
    let socket2 = tonic::transport::server::TcpIncoming::new("127.0.0.1:8080".parse().unwrap(), true, None);
}
```

The code can run successfully.

Intuitively, binding an address may involve IO, and the method should be async. But before `server.await` there is no await.

Reply:
> The code can run successfully.

Is that because `TcpIncoming::new` returns `Result`? 😄

Reply:
My bad 🤡